### PR TITLE
Update `qml.wires.Wires` to disallow `wires=None`

### PR DESCRIPTION
**Pull request by andrijapau**
**Context:**

Currently, the behaviour is,
```python
>>> qml.wires.Wires(None)
Wires([None])
```
which doesn't make sense.

With this change we get the intended behaviour,
```python
>>> qml.wires.Wires(None)
TypeError: Must specify a set of wires.
```

**Description of the change:**

Had to clean-up some templates that relied on defaulting to `None` 😓. 

**Benefits:** Logical code 😃 

**Possible Drawbacks:** None

[sc-80454]

### CODE REVIEW

**Review by KetpuntoG — COMMENTED** [review]
Looks good! 
Wouldn't be more beneficial to make sure `Wires([None])` is not a valid wire?
If that was solved, this would not be necessary

**Comment by andrijapau** [comment]
@KetpuntoG, great question. The decision to move away from `wires=None` was recently escalated due to the problems it is causing in the program capture initiative. Additionally, having `wires` set to `None` (or `Wires([None])` doesn't really make sense) so this was a good opportunity to clean up the codebase.

**Review by KetpuntoG — APPROVED** [review]
Thanks!

File: pennylane/wires.py [hunk 1]
```diff
@@ -129,6 +129,8 @@ def _unflatten(cls, data, _metadata):
         return cls(data, _override=True)
 
     def __init__(self, wires, _override=False):
+        if wires is None:
+            raise ValueError("Must specify a set of wires.")
```
Comment (albi3ro) [inline pennylane/wires.py:133]:
```suggestion
            raise ValueError("Must specify a set of wires. None is not a valid wire label.")
```
Maybe? I feel like that phrasing is a little passive aggressive, but I'm thinking we should communicate something more explicit about having received `None` and it not being an acceptable wire label.

File: doc/releases/changelog-dev.md [hunk 2]
```diff
@@ -447,6 +447,9 @@ same information.
 
 <h3>Bug fixes 🐛</h3>
 
+* The `Wires` object throws a `ValueError` if `wires=None`.
```
[the remark this answers is not in the document]
Comment (andrijapau) [reply doc/releases/changelog-dev.md:450]: Hmmm. True?

[the remark this answers is not in the document]
Comment (andrijapau) [reply doc/releases/changelog-dev.md:450]: Yeah, I agree.

Comment (astralcai) [inline doc/releases/changelog-dev.md:450]:
```suggestion
* The `Wires` object throws a `TypeError` if `wires=None`. 
```

**Review by astralcai — APPROVED** [review]

**Review by albi3ro — APPROVED** [review]
Looks great! thanks for making it so the operators still accept `None` as an indication for "no wires".

File: tests/test_wires.py [hunk 3]
```diff
@@ -34,6 +34,11 @@
 class TestWires:
     """Tests for the ``Wires`` class."""
 
+    def test_error_if_wires_none(self):
+        """Tests that a ValueError is raised if None is given as wires."""
```
Comment (andrijapau) [inline tests/test_wires.py:38]:
```suggestion
        """Tests that a TypeError is raised if None is given as wires."""
```

**Pull request merged**